jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/websocket/WebSocketDiagnosticsCollector.java

```diff
@@ -8,30 +8,196 @@
+                        }
+
+                        // check parameters valid types
+                        if (!(isSpecialType || isPrimWrapped || isPrimitive)) {
+                            Diagnostic diagnostic = createDiagnostic(param, unit,
+                                    WebSocketConstants.DIAGNOSTIC_ON_OPEN_INVALID_PARAMS,
+                                    WebSocketConstants.DIAGNOSTIC_CODE_ON_OPEN_INVALID_PARAMS);
+                            diagnostics.add(diagnostic);
+                            return;
+                        }
+
+                        if (!isSpecialType) {
+                            // check that if parameter is not a specialType, it has a @PathParam annotation
+                            IAnnotation[] param_annotations = param.getAnnotations();
+                            boolean hasPathParamAnnot = Arrays.asList(param_annotations).stream().anyMatch(
+                                    annot -> annot.getElementName().equals(WebSocketConstants.PATH_PARAM_ANNOTATION));
+
+                            if (!hasPathParamAnnot) {
+                                Diagnostic diagnostic = createDiagnostic(param, unit,
+                                        WebSocketConstants.DIAGNOSTIC_PATH_PARAMS_ANNOT_MISSING,
+                                        WebSocketConstants.DIAGNOSTIC_CODE_PATH_PARMS_ANNOT);
+                                diagnostics.add(diagnostic);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+    * Check if valueClass is a wrapper object for a primitive value
+    * Based on https://github.com/eclipse/lsp4mp/blob/9789a1a996811fade43029605c014c7825e8f1da/microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/utils/JDTTypeUtils.java#L294-L298
+    * 
+    * @param valueClass the resolved type of valueClass in string or the simple type of valueClass 
+    * @return if valueClass is a wrapper object
+    */
+    private boolean isWrapper(String valueClass) {
+        return WebSocketConstants.WRAPPER_OBJS.contains(valueClass) || WebSocketConstants.RAW_WRAPPER_OBJS.contains(valueClass);
+    }
+
+
+    /**
+    * Checks if type is a WebSocket endpoint by meeting one of the 2 conditions listed on 
+    * https://jakarta.ee/specifications/websocket/2.0/websocket-spec-2.0.html#applications 
+    * are met: class is annotated or class implements Endpoint class
+    * 
+    * @param type the type representing the class
+    * @return the conditions for a class to be a WebSocket endpoint 
+    * @throws JavaModelException
+    */
+    private HashMap<String, Boolean> isWSEndpoint(IType type) throws JavaModelException {
+        HashMap<String, Boolean> wsEndpoint = new HashMap<>();
+
+        // check trivial case
+        if (!type.isClass()) {
+            wsEndpoint.put("isAnnotation", false);
```

Comment:
When you use a string as a key like this it is prone to bugs. Someone could accidentally code `.get("isAnotation")` and introduce a bug which may not be caught. Please put this string in a named constant so that the compiler can check for a misspelt variable name.

Reply:
Makes sense, thanks! I just pushed a new change handling it this